### PR TITLE
HPCC-12943 Added LD_PRELOAD export to hpcc_setenv

### PR DESCRIPTION
**Pull request by Michael-Gardner**
Signed-off-by: Michael Gardner Michael.Gardner@lexisnexis.com

### CODE REVIEW

**Comment by Michael-Gardner** [comment]
@xwang2713 @rpastrana Please review/test when you have time.

File: initfiles/sbin/hpcc_setenv.in [hunk 1]
```diff
@@ -81,6 +81,7 @@ IFS="${OIFS}"
 PATH_PREFIX=`cat ${HPCC_CONFIG} | sed -n "/\[${SECTION}\]/,/\[/p" | grep "^path *= *" | sed -e 's/^path *= *//'`
 
 export PID=`cat ${HPCC_CONFIG} | sed -n "/\[${SECTION}\]/,/\[/p" | grep "^pid *= *" | sed -e 's/^pid *= *//'`
+export LD_PRELOAD=`cat ${HPCC_CONFIG} | sed -n "/\[${SECTION}\]/,/\[/p" | grep "^java *= *" | sed -e 's/^java *=*//'`
```
Comment (richardkchapman) [inline initfiles/sbin/hpcc_setenv.in]: This will override any existing LD_PRELOAD - is that desirable?

**Comment by Michael-Gardner** [comment]
@richardkchapman went ahead and made it so the LD_PRELOAD variable won't get overwritten, and rebased to fix merge issues with environment.conf.in

File: initfiles/sbin/hpcc_setenv.in [hunk 2]
```diff
@@ -81,6 +81,7 @@ IFS="${OIFS}"
 PATH_PREFIX=`cat ${HPCC_CONFIG} | sed -n "/\[${SECTION}\]/,/\[/p" | grep "^path *= *" | sed -e 's/^path *= *//'`
 
 export PID=`cat ${HPCC_CONFIG} | sed -n "/\[${SECTION}\]/,/\[/p" | grep "^pid *= *" | sed -e 's/^pid *= *//'`
+export LD_PRELOAD=`cat ${HPCC_CONFIG} | sed -n "/\[${SECTION}\]/,/\[/p" | grep "^java *= *" | sed -e 's/^java *=*//'`:$LD_PRELOAD
```
Comment (richardkchapman) [inline initfiles/sbin/hpcc_setenv.in]: Should the sed command have a space after the = (like the grep does) ?

**Comment by richardkchapman** [comment]
@Michael-Gardner back to you

File: initfiles/sbin/hpcc_setenv.in (hunk 2 above)
Comment (Michael-Gardner) [inline initfiles/sbin/hpcc_setenv.in]: I actually think it should probably be like that for the pid as well.  Going to test it and make sure.

**Comment by Michael-Gardner** [comment]
@richardkchapman changed the LD_PRELOAD line to mimic the PID line above it, and match zero or more instances of ' ' for the sed call.

File: initfiles/sbin/hpcc_setenv.in [hunk 3]
```diff
@@ -81,6 +81,7 @@ IFS="${OIFS}"
 PATH_PREFIX=`cat ${HPCC_CONFIG} | sed -n "/\[${SECTION}\]/,/\[/p" | grep "^path *= *" | sed -e 's/^path *= *//'`
 
 export PID=`cat ${HPCC_CONFIG} | sed -n "/\[${SECTION}\]/,/\[/p" | grep "^pid *= *" | sed -e 's/^pid *= *//'`
+export LD_PRELOAD=`cat ${HPCC_CONFIG} | sed -n "/\[${SECTION}\]/,/\[/p" | grep "^java *= *" | sed -e 's/^java *= *//'`:$LD_PRELOAD
```
Comment (mckellyln) [inline initfiles/sbin/hpcc_setenv.in]: Should we test first and then set (or add to) LD_PRELOAD only if this is not empty ?
Probably does not matter but don't see a need to force LD_PRELOAD always, even if empty.
Something like:
jvm_preload=$(cat ...)
if [ -n "${jvm_preload}" ] ; then
  export LD_PRELOAD=${jvm_preload}:${LD_PRELOAD}
fi

Comment (Michael-Gardner) [inline initfiles/sbin/hpcc_setenv.in]: I just double checked to make sure.  Having just "java=" in the conf file has no adverse effects.  The result will be ":" or ":/whatever/ld_preload/was.so" and both are handled fine by the system.

**Pull request merged**